gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/ParquetGroup.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.parquet;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import parquet.example.data.Group;
+import parquet.example.data.simple.BinaryValue;
+import parquet.example.data.simple.BooleanValue;
+import parquet.example.data.simple.DoubleValue;
+import parquet.example.data.simple.FloatValue;
+import parquet.example.data.simple.Int96Value;
+import parquet.example.data.simple.IntegerValue;
+import parquet.example.data.simple.LongValue;
+import parquet.example.data.simple.NanoTime;
+import parquet.example.data.simple.Primitive;
+import parquet.io.api.Binary;
+import parquet.io.api.RecordConsumer;
+import parquet.schema.GroupType;
+import parquet.schema.PrimitiveType;
+import parquet.schema.Type;
+
+import static parquet.schema.Type.Repetition.REPEATED;
+
+
+/**
+ * Custom Implementation of {@link Group} to support adding {@link Object} of type {@link Primitive} or {@link Group}.
+ * Also provides methods to add {@link Primitive} and {@link Group} with {@link String} key if index is not known.
+ * @author tilakpatidar
+ */
+public class ParquetGroup extends Group {
+
+  private final GroupType schema;
+  //data indexed by fieldIndex of the schema
```

Comment:
The important message here is this each list item represents data of a field, which is indexed by the fieldIndex of the schema. Could you make it clearer?